Scripts/Battle/battle_manager.gd

```diff
@@ -454,23 +506,23 @@ class AnimInQueue:
 		result.debug_name = "turn_"+unit_form_.unit.template.unit_name
 		result._unit_form = unit_form_
 		unit_form_.anim_end.connect(result.on_anim_end)
-		result._animate = func () : unit_form_.start_turn_anim()
+		result._animate = func () : if (unit_form_): unit_form_.start_turn_anim()
```

Comment:
why is this needed?
if you try animating null unit_form_ it will break on not getting to anim end ever
null check doesn't help much

Reply:
It is needed when we quit battle when any animation is playing -- the clean up i added is not sufficient and we need to do additional null check